CloudOnMobile/CommandManager.swift

```diff
@@ -45,31 +31,51 @@ final class CommandManager {
         var payload: T?
     }
 
-    func connect() {
-        Thread.detachNewThread {
+    weak var delegate: CommandManagerDelegate?
+
+    private var code: UInt32 = 0
+
+    private var isConnected = false
+
+    private let ip: String
+
+    private let port: Int = 9293
+
+    private let documentsDirectory: URL
+
+    init(url: String) {
+        ip = url
+        initSharedCore()
+        let fileMngr = FileManager.default
+        documentsDirectory = fileMngr.urls(for: .documentDirectory, in: .userDomainMask)[0]
+        setup_environment(documentsDirectory.path)
+    }
+
+    func connect() async throws -> Int {
```

Comment:
This function shouldn't be `async` because it never returns. `self.endlessListen()` doesn't return, so the whole function won't return.